packages/dashboard/src/components/userPropertiesTable.tsx

```diff
@@ -0,0 +1,215 @@
+  filterable: false,
+  renderCell: monospaceCell,
+};
+
+export default function UserPropertiesTable() {
+  const router = useRouter();
+
+  const userPropertiesResult = useAppStore((store) => store.userProperties);
+  const userProperties =
+    userPropertiesResult.type === CompletionStatus.Successful
+      ? userPropertiesResult.value
+      : [];
+
+  const usersPropertiesRow: Row[] = [];
+
+  userProperties.forEach((userProperty) => {
+    const row: Row = {
+      id: userProperty.id,
+      properties: userProperty.name,
+      updatedAt: userProperty.updatedAt
+        ? new Date(userProperty.updatedAt).toISOString()
+        : "Not Updated",
+      templates: userProperty.templates ?? "No Templates",
+      lastRecomputed: userProperty.lastRecomputed
+        ? new Date(userProperty.lastRecomputed).toISOString()
+        : "Not Re-Computed",
+    };
+    usersPropertiesRow.push(row);
+  });
+
+  const setUserPropertyDeleteRequest = useAppStore(
+    (store) => store.setUserPropertyDeleteRequest,
+  );
+  const apiBase = useAppStore((store) => store.apiBase);
+  const userPropertyDeleteRequest = useAppStore(
+    (store) => store.userPropertyDeleteRequest,
+  );
+  const deleteUserProperty = useAppStore((store) => store.deleteUserProperty);
+
+  const setDeleteResponse = (
+    _response: EmptyResponse,
+    deleteRequest?: DeleteUserPropertyRequest,
+  ) => {
+    if (!deleteRequest) {
+      return;
+    }
+    deleteUserProperty(deleteRequest.id);
+  };
+
+  return (
+    <DataGrid
+      rows={usersPropertiesRow}
+      sx={{
+        height: "100%",
+        width: "100%",
+        // disable cell selection style
+        ".MuiDataGrid-cell:focus": {
+          outline: "none",
+        },
+        // pointer cursor on ALL rows
+        "& .MuiDataGrid-row:hover": {
+          cursor: "pointer",
+        },
+      }}
+      getRowId={(row) => row.id}
+      onRowClick={(params) => {
+        router.push({
+          pathname: `/user-properties/${params.id}`,
+        });
+      }}
+      autoPageSize
+      columns={[
+        {
+          field: "properties",
+          headerName: "Name",
+        },
+        {
+          field: "lastRecomputed",
+          headerName: "Last Re-Computed",
+        },
+        {
+          field: "updatedAt",
+          headerName: "Last Updated",
+        },
+        {
+          field: "templates",
+          headerName: "Templates used by",
+          renderCell: ({ row }: { row: Row }) => {
+            const currentRow = row;
+            if (currentRow.templates === "No Templates") {
+              return (
+                <div>
+                  <p>No Templates</p>
+                </div>
+              );
+            }
+            const templates = currentRow.templates.split(",");
+            return (
+              <div>
+                {templates.map((template) => {
+                  const [templateName, templateId, templateType] =
+                    template.split("|");
```

Comment:
This is really unconventional code. Rather than joining and then splitting strings, you should just have an object that contains  `{ templateName: string, templateId: string, templateType: ChannelType }`.